### ISSUE TITLE
False positives on `no-internal-module`

### ISSUE THREAD

**Issue by jiaweihli** [palantir/tslint#679]
```typescript
/// <reference path="../../../../../typings/tsd.d.ts" />

module MyModule {}

export default MyModule;
```

is triggering `no-internal-module`.

**Comment by JKillian** [comment]
The intent of the `no-internal-module` rule, if I understand right, is simply about favoring the `namespace` keyword instead of the `module` keyword for internal modules. So if you change your code to read `namespace MyModule {}` the error will go away.

**Comment by jiaweihli** [comment]
I'm exporting the module, and importing it in the style of external modules from other files.

**Comment by jiaweihli** [comment]
Can this be re-opened until the issue is clarified?

**Comment by JKillian** [comment]
We're probably not going to make a special case for this. Is there a reason you're using modules/namespaces inside an external module? one of the TypeScript devs [recommends against this](http://stackoverflow.com/questions/30357634/how-do-i-use-namespaces-with-typescript-external-modules) unless you are exporting many things from your external module.

Either way, you can change `module` to `namespace` and everything will work identically, except the error will go away.

**Comment by adidahiya** [comment]
@jiaweihli do you still want to keep this open? `namespace` is a direct replacement for `module`, and the spirit of the `no-internal-module` rule is to encourage usage of the newer, more semantically correct keyword. Perhaps it should've been called `no-module-keyword`, but I don't see much benefit in changing that now.

**Comment by jiaweihli** [comment]
Modules will continue to exist for the purpose of external modules: https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Namespaces%20and%20Modules.md#a-note-about-terminology

`namespace` only replaces internal modules, but not external ones.

Re: http://stackoverflow.com/a/30357635/759714, my use case is the last one - I export several things, and I want to group them under an umbrella name to prevent top-level conflicts with other imports.

I don't have much code that conflicts with this, so I could probably refactor it into an abstract class with static methods (though that makes less sense).  Renaming the keyword would make things much clearer, though I understand that probably won't happen until v3.0 at least.

**Comment by Maks3w** [comment]
I have this warning with definition files (`.d.ts`) I think this files should be excluded from this inspection.

Basically I follow the same conventions of DefinitelyTyped

**Comment by adidahiya** [comment]
Upon reconsideration, I don't think `no-module-keyword` is the right name either, because the rule permits external modules with that keyword (`export module "tslint" { ... }`). I won't be renaming this for 3.0.
